riscv/src/compiler.rs

```diff
@@ -73,6 +73,125 @@ machine Binary(latch, function_id) {
         {function_id', A_byte, B_byte, C_byte} in {P_operation, P_A, P_B, P_C};
     }
 }
+"#,
+        r#"
+        // Implements the poseidon permutation for the BN254 curve.
+        // Note that this relies on the trace table being non-wrapping, so it will
+        // only work with the Halo2 backend (which is the only backend that supports
+        // the BN254 curve).
+        machine PoseidonBN254(LASTBLOCK, function_id) {
```

Comment:
We could also have this as a separate pil file and test it, also before merging this.

Reply:
That's done! I copied this from test_data/asm/poseido_bn254.asm which is merged and tested already